src/quickfix.c

```diff
@@ -1363,7 +1363,8 @@ qf_parse_multiline_pfx(
 	    qfprev->qf_lnum = fields->lnum;
 	if (!qfprev->qf_col)
 	    qfprev->qf_col = fields->col;
-	qfprev->qf_viscol = fields->use_viscol;
+        if (!qfprev->qf_viscol)
+            qfprev->qf_viscol = fields->use_viscol;
```

Comment:
The qf_viscol field should be set whenever the qf_col field is set. So the
change should look like:

````
if (!qfprev->qf_col)
{
    qfprev->qf_col = fields->col;
    qfprev->qf_viscol = fields->use_viscol;
}
````

Can you also add a new test in test_quickfix.vim for this?

Reply:
> The qf_viscol field should be set whenever the qf_col field is set. So the
change should look like: ...

Thanks! Will try that out.

> Can you also add a new test in test_quickfix.vim for this?

I'll do my best. Need to learn about the test infrastructure, so it may take some time.

Reply:
You can add the following lines to the Test_viscol() function in the
testdir/test_quickfix.vim file (after line 4425):

```
  " Use screen column number with a multi-line error message
  enew
  call writefile(["à errormsg"], 'Xfile1')
  set efm=%E===\ %f\ ===,%C%l:%v,%Z%m
  cexpr ["=== Xfile1 ===", "1:3", "errormsg"]
  call assert_equal('Xfile1', @%)
  call assert_equal([0, 1, 4, 0], getpos('.'))
```